protocol/contracts/templegold/TempleGoldStaking.sol

```diff
@@ -0,0 +1,489 @@
+        uint256 _prevBalance = _balances[_for];
+        _applyStake(_for, _amount);
+        _mintVoteToken(_for, _amount);
+        _updateAccountWeight(_for, _prevBalance, _balances[_for], true);
+    }
+
+    /**
+     * @notice Withdraw staked tokens
+     * @param amount Amount to withdraw
+     * @param claim Boolean if to claim rewards
+     */
+    function withdraw(uint256 amount, bool claim) external override {
+        _withdrawFor(msg.sender, msg.sender, amount, claim, msg.sender);
+    }
+
+    /**
+     * @notice Withdraw all staked tokens
+     * @param claim Boolean if to claim rewards
+     */
+    function withdrawAll(bool claim) external override {
+        _withdrawFor(msg.sender, msg.sender, _balances[msg.sender], claim, msg.sender);
+    }
+
+    /// @notice Owner can pause user swaps from occuring
+    function pause() external override onlyElevatedAccess {
+        _pause();
+    }
+
+    /// @notice Owner can unpause so user swaps can occur
+    function unpause() external override onlyElevatedAccess {
+        _unpause();
+    }
+
+    /**
+     * @notice Get account staked balance
+     * @param account Account
+     * @return Staked balance of account
+     */
+    function balanceOf(address account) public override view returns (uint256) {
+        return _balances[account];
+    }
+
+    /**
+     * @notice Get earned rewards of account
+     * @param account Account
+     * @return Earned rewards of account
+     */
+    function earned(address account) external override view returns (uint256) {
+        return _earned(account, _balances[account]);
+    }
+
+    /**
+     * @notice Get Temple Gold reward per token of Temple 
+     * @return Reward per token
+     */
+    function rewardPerToken() external override view returns (uint256) {
+        return _rewardPerToken();
+    }
+
+    /**
+     * @notice Get finish timestamp of rewards
+     * @return Finish timestamp
+     */
+    function rewardPeriodFinish() external view returns (uint40) {
+        return rewardData.periodFinish;
+    }
+
+    /**
+     * @notice Elevated access can recover tokens which are not staking or reward tokens
+     * @param _token Token to recover
+     * @param _to Recipient
+     * @param _amount Amount of tokens
+     */
+    function recoverToken(address _token, address _to, uint256 _amount) external override onlyElevatedAccess {
+        if (_token == address(stakingToken) || _token == address(rewardToken )) { revert CommonEventsAndErrors.InvalidAddress(); }
+
+        IERC20(_token).safeTransfer(_to, _amount);
+        emit CommonEventsAndErrors.TokenRecovered(_to, _token, _amount);
+    }
+
+    /**  
+     * @notice Get rewards
+     * @param staker Staking account
+     */
+    function getReward(address staker) external override updateReward(staker) {
+        _getReward(staker, staker);
+    }
+
+    /**  
+     * @notice Get vote weight of an account
```

Comment:
Are these weights actually needed/used though? Because you have the `voteToken.mint()`.

What's the difference in this vote weight vs the minted vote tokens?

If not needed, should remove to get rid of some of the complexity?

Reply:
discussed in discord. will mark resolved